tensorboard/plugins/debugger/debugger_server_test.py

```diff
@@ -0,0 +1,260 @@
+  keeps track of events that would be written to disk.
+  """
+
+  def __init__(self, events_output_list):
+    """Constructs a fake events writer, which appends events to a list.
+
+    Args:
+      events_output_list: The list to append events that would be written to
+        disk.
+    """
+    self.events_written = events_output_list
+
+  def dispose(self):
+    """Does nothing. This implementation creates no file."""
+
+  def write_event(self, event):
+    """Pretends to write an event to disk.
+
+    Args:
+      event: The event proto.
+    """
+    self.events_written.append(event)
+
+
+class DebuggerDataServerTest(tf.test.TestCase):
+
+  def setUp(self):
+    self.events_written = []
+
+    events_writer_manager = FakeEventsWriterManager(self.events_written)
+    self.stream_handler = debugger_server_lib.DebuggerDataStreamHandler(
+        events_writer_manager=events_writer_manager)
+    self.stream_handler.on_core_metadata_event(tf.Event())
+
+  def tearDown(self):
+    mock.patch.stopall()
+
+  def _create_event_with_float_tensor(self, node_name, output_slot, debug_op,
+                                      list_of_values):
+    """Creates event with float64 (double) tensors.
+
+    Args:
+      node_name: The string name of the op. This lacks both the output slot as
+        well as the name of the debug op.
+      output_slot: The number that is the output slot.
+      debug_op: The name of the debug op to use.
+      list_of_values: A python list of values within the tensor.
+    Returns:
+      A `tf.Event` with a summary containing that node name and a float64
+      tensor with those values.
+    """
+    event = tf.Event()
+    value = event.summary.value.add()
+    value.tag = node_name
+    value.node_name = "%s:%d:%s" % (node_name, output_slot, debug_op)
+    value.tensor.dtype = tf.float64.as_datatype_enum
+    value.tensor.tensor_shape.dim.add().size = len(list_of_values)
+    value.tensor.tensor_content = numpy.array(
+        list_of_values, dtype=numpy.float64).tobytes()
+    plugin_content = debugger_event_metadata_pb2.DebuggerEventMetadata(
+        device="/job:localhost/replica:0/task:0/cpu:0", output_slot=output_slot)
+    value.metadata.plugin_data.add(
+        plugin_name=constants.DEBUGGER_PLUGIN_NAME,
+        content=json_format.MessageToJson(
+            plugin_content, including_default_value_fields=True))
+    return event
+
+  def _verify_event_lists_have_same_tensor_values(self, expected, gotten):
+    """Checks that 2 lists of events have the same tensor values.
+
+    Args:
+      expected: The expected list of events.
+      gotten: The list of events we actually got.
+    """
+    self.assertEqual(len(expected), len(gotten))
+
+    # Compare the events one at a time.
+    for expected_event, gotten_event in zip(expected, gotten):
+      self.assertEqual(expected_event.summary.value[0].node_name,
+                       gotten_event.summary.value[0].node_name)
+      self.assertEqual(
```

Comment:
Not sure what the status of this is. The code that I want to see is of the form:
```py
self.assertAllClose(
    tf.make_ndarray(expected_event.summary.value[0].tensor),
    tf.make_ndarray(actual_event.summary.value[0].tensor))
```
It is **not safe** to assume that if two tensor protos are different then they represent different tensors. This is true both in theory and in practice: for instance, if [this function in `scalar/summary_test.py`](https://github.com/tensorflow/tensorboard/blob/615eccdaa5119d2c3aaf4870f5882fa2e606b47c/tensorboard/plugins/scalar/summary_test.py#L39) is replaced with the identity function, then the test **fails** on TensorFlow 1.3.